### PR TITLE
Feature/modulize blog

### PR DESCRIPTION
**Pull request by poespas**

### CODE REVIEW

**Comment by dlindeboom** [comment]
Zou je mij kunnen uitleggen wat dit precies oplost deze pr?

**Comment by poespas** [comment]
@dlindeboom

This PR (re)moves all blog-related things to it's own private module, creates a gernalized Wordpress HTTPClient so we can easily inherit from that, and generally clean up the code.

File: Model/HttpClient/Wordpress.php [hunk 1]
```diff
@@ -8,14 +8,15 @@
 use Symfony\Component\HttpClient\HttpClient;
 use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
 
-class Post
+class Wordpress
 {
     const WP_JSON_URL_PREFIX = '/wp-json/wp/v2/';
+    public const WP_JSON_URL_SUFFIX = '';
```
[the remark this answers is not in the document]
Comment (dlindeboom) [reply Model/HttpClient/Wordpress.php:14]: Like talked about, we need to make this class abstract and dont use a const

**Review by dlindeboom — CHANGES_REQUESTED** [review]

File: Model/HttpClient/Wordpress.php [hunk 2]
```diff
@@ -48,8 +50,11 @@ public function setBaseUrl(string $baseUrl)
      * @throws \Symfony\Contracts\HttpClient\Exception\ServerExceptionInterface
      * @throws \Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface
      */
-    public function all(int $pageSize = 10): \Generator
+    public function all(int $pageSize = 10, array $filters = []): \Generator
     {
+        if (is_null($this->type)) {
+            throw new \Exception("Type is required for HTTPClient");
```
Comment (dlindeboom) [inline Model/HttpClient/Wordpress.php:56]: Even import van maken en een eigen exception

**Review by dlindeboom — CHANGES_REQUESTED** [review]

**Review by dlindeboom — APPROVED** [review]

**Pull request merged**